test/query_property_test.dart

```diff
@@ -0,0 +1,406 @@
+    final queryStrings  = box.query(tString.endsWith("suffix")).build();
+
+    final start = [ 2, 4, 5 ];
+    for (int i=0; i<tIntegers.length; i++) {
+      final qp = queryIntegers.property(tIntegers[i]) as IntegerPropertyQuery;
+
+      final mappedIntegers = integers.map((j) => j + start[i]).toList();
+      expect(qp.find(defaultValue:-1), mappedIntegers);
+      expect(qp.find(), mappedIntegers);
+
+      qp.close();
+    }
+
+    tFloats.forEach((f) {
+      final qp = queryFloats.property(f) as DoublePropertyQuery;
+
+      double d = tFloat == f ? 0.20000000298023224 : 0.30000000000000004;
+
+      expect(qp.find(), [ d, d, d, d, d ]);
+      expect(qp.find(defaultValue:-0.1), [ d, d, d, d, d ]);
+
+      qp.close();
+    });
+
+    final qp = queryStrings.property(tString) as StringPropertyQuery;
+
+    final defaultResult = ['1withSuffix', '2withSuffix', '1withsuffix', '2withsuffix'];
+    expect(qp.find(), defaultResult);
+    expect((qp..distinct = true ..caseSensitive = true) .find(), ['2withsuffix', '1withsuffix', '2withSuffix', '1withSuffix'] );
+    expect((qp..distinct = false..caseSensitive = true) .find(defaultValue:"meh"), ['1withSuffix', '2withSuffix', '1withsuffix', '2withsuffix']);
+    expect((qp..distinct = true ..caseSensitive = false).find(), ['2withSuffix', '1withSuffix']);
+    expect((qp..distinct = false..caseSensitive = false).find(defaultValue:"meh"), defaultResult);
+    qp.close();
+
+    queryIntegers.close();
+    queryFloats.close();
+    queryStrings.close();
+  });
+
+  test(".average", () {
+    box.putMany(integerList);
+    box.putMany(floatList);
+
+    final queryIntegers = box.query(tLong.lessThan(1000)).build();
+    final queryFloats   = box.query(tDouble.lessThan(1000.0)).build();
+
+    // integers
+    double intBaseAvg = integers.reduce((a,b) => a + b) / integers.length;
+
+    final qpInteger = (p, avg) {
+      final qp = queryIntegers.integerProperty(p);
+      expect(qp.average(), avg);
+      qp.close();
+    };
+
+    qpInteger(tLong, intBaseAvg + 5);
+    qpInteger(tInt, intBaseAvg + 4);
+    qpInteger(tShort, intBaseAvg + 2);
+
+    // floats
+    final qpFloat = (p, avg) {
+      final qp = queryFloats.doubleProperty(p);
+      expect(qp.average(), avg);
+      qp.close();
+    };
+
+    double floatBaseAvg = floats.reduce((a,b) => a + b) / floats.length;
+
+    qpFloat(tFloat,  floatBaseAvg + 0.10000000298023224);
+    qpFloat(tDouble, 0.30000000000000004);
+
+    // close
+    queryFloats.close();
+    queryIntegers.close();
+  });
+
+  test(".find() default values on null results" , () {
+    box.putMany(integerList);
+    box.putMany(stringList);
+    box.putMany(floatList);
+
+    final queryIntegers = box.query(tLong.greaterThan(1000)).build();
+    final queryFloats   = box.query(tDouble.greaterThan(1000.0)).build();
+    final queryStrings  = box.query(tString.equals("can't find this")).build();
+
+    final integerDefaultValue = -2;
```

Comment:
this is misunderstood, the default value will be used in place of `NULL` on a given property, i.e you need to first insert objects with NULLs.

Reply:
Ah, thanks for clarifying that.

Reply:
I updated the param name and tests to reflect that.